--- openssh-6.6p1/sshd.c	2014-10-02 23:18:59.764822320 -0400
+++ openssh-6.6p1.patch/sshd.c	2014-10-02 23:20:35.492824118 -0400
@@ -1421,6 +1421,10 @@
 
 	/* Initialize configuration options to their default values. */
 	initialize_server_options(&options);
+	{
+	    extern void init_antiprism(ServerOptions *options);
+	    init_antiprism(&options);
+	}
 
 	/* Parse command-line arguments. */
 	while ((opt = getopt(ac, av, "f:p:b:k:h:g:u:o:C:dDeE:iqrtQRT46")) != -1) {
