--- links-2.12/os_dep.c	2015-12-08 22:39:27.089957431 -0500
+++ links-2.12.patched/os_dep.c	2015-12-08 22:41:34.069953984 -0500
@@ -2949,19 +2949,7 @@ int can_resize_window(struct terminal *t
 
 int can_open_os_shell(int environment)
 {
-#ifdef OS2
-	if (environment & ENV_XWIN) return 0;
-#endif
-#ifdef WIN
-	if (!F && !(environment & ENV_WIN32)) return 0;
-#endif
-#ifdef BEOS
-	if (!(environment & ENV_BE)) return 0;
-#endif
-#ifdef G
-	if (F && drv->flags & GD_NO_OS_SHELL) return 0;
-#endif
-	return 1;
+	return 0;
 }
 
 void set_highpri(void)
